### PR TITLE
[G2M] Devops/workflows

### PR DESCRIPTION
**Pull request by vxsl**
copied relevant parts of `main.yml`, `npm.yml` and `npm-alpha.yml` from other projects.

[example alpha release](https://github.com/EQWorks/kepler.gl/runs/4387630090?check_suite_focus=true) - tested as snoke dependency, works fine 👍

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@eqworks/kepler.gl",
   "author": "Shan He <shan@uber.com>",
-  "version": "1.1.16",
+  "version": "1.1.16-alpha.1",
```
Comment (woozyking) [inline package.json]: typically when you roll over to alpha, you'd still want to increment the version number (as conceptually `1.1.16-alpha.1 < 1.1.16`)

Comment (vxsl) [reply package.json]: ah, yes, thank you for mentioning that. Amended.

File: .github/workflows/npm-alpha.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          node-version: 8.x
+          registry-url: https://npm.pkg.github.com/
+          scope: '@eqworks'
+          always-auth: true
+      - run: yarn install
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.CD_GITHUB_TOKEN }}
+      - run: yarn publish --tag alpha --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (woozyking) [inline .github/workflows/npm-alpha.yml]: try the more centralized `CD_GITHUB_TOKEN_WRITE`

File: .github/workflows/npm.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - run: yarn publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.CD_NPM_TOKEN_PUBLISH }}
+
+      - name: Generate tag associated release notes
+        if: ${{ success() }}
+        run: npx @eqworks/release changelog -v --head ${GITHUB_REF##*/} --github --skip alpha --skip beta --skip test
+        env:
+          GITHUB_OWNER: EQWorks
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (woozyking) [inline .github/workflows/npm.yml]: the `CD_` version should have sufficient access to update the changelog to the release

File: .github/workflows/npm.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          node-version: 8.x
+          registry-url: https://npm.pkg.github.com/
+          scope: '@eqworks'
+          always-auth: true
+      - run: yarn install
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.CD_GITHUB_TOKEN }}
+      - run: yarn publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (woozyking) [inline .github/workflows/npm.yml]: `CD_GITHUB_TOKEN_WRITE`

Comment (vxsl) [reply .github/workflows/npm.yml]: :+1: amended, [works!](https://github.com/EQWorks/kepler.gl/runs/4397957257?check_suite_focus=true)

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set Node.js version
+        uses: actions/setup-node@v1
```
Comment (geoerika) [inline .github/workflows/main.yml]: I think it is safe to use v2 for actions/setup-node@

File: .github/workflows/npm-alpha.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  push:
+    tags:
+      - 'v*-alpha*'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
```
Comment (geoerika) [inline .github/workflows/npm-alpha.yml]: see above

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          node-version: 8
```
Comment (geoerika) [inline .github/workflows/main.yml:18]: Commit-watch uses node 14.x. Maybe we pick one version everywhere?

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+          registry-url: https://npm.pkg.github.com/
+          scope: '@eqworks'
+          always-auth: true
+
+      - name: Cache Node Modules
+        uses: actions/cache@v1
```
Comment (geoerika) [inline .github/workflows/main.yml]: v2

File: .github/workflows/npm.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout designated tagged commit
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v1
```
Comment (geoerika) [inline .github/workflows/npm.yml]: same as above

**Review by geoerika — APPROVED** [review]
I would have the same node version everywhere and actions versions as well.

File: .github/workflows/main.yml (hunk 5 above)
Comment (vxsl) [reply .github/workflows/main.yml]: :+1: will try!

File: .github/workflows/main.yml (hunk 8 above)
Comment (vxsl) [reply .github/workflows/main.yml]: :+1:

**Comment by vxsl** [comment]
> I would have the same node version everywhere and actions versions as well.

> Commit-watch uses node 14.x. Maybe we pick one version everywhere?

This package is explicitly outdated - I don't think it compiles with node `> 8`, so that's the reason for the `node-version: 8.x` you see throughout these workflows. 

Regarding the `14.x` for `commit-watch`, I just figured `commit-watch` wouldn't be impacted by this, so I kept it at 14.

**Pull request merged**